### ISSUE TITLE
Tests wait on all nodes to finish before continuing next group

### ISSUE THREAD

**Issue by spursbob** [pytest-dev/pytest-xdist#453]
In the image you can see one test is taking a long time to complete while 23 nodes sit waiting for that test to finish. Is there a way to prevent this?

![image](https://user-images.githubusercontent.com/44440300/61631089-cfd19400-ac5f-11e9-8b5c-2a8df34dfec6.png)

**Comment by Dude-X** [comment]
There isn't at the moment. It requires a change to pytest core where the iteration of tests can be done in a better manner.

It irks me too, that it practically batches tests, instead of queueing them up.

**Comment by martinpriestley** [comment]
This is a duplicate of #339.
